presto-main/src/main/java/com/facebook/presto/sql/analyzer/ScopeReferenceExtractor.java

```diff
@@ -38,20 +36,18 @@ public static boolean hasReferencesToScope(Node node, Analysis analysis, Scope s
 
     public static Stream<Expression> getReferencesToScope(Node node, Analysis analysis, Scope scope)
     {
-        Map<NodeRef<Expression>, FieldId> columnReferences = analysis.getColumnReferenceFields();
+        ListMultimap<NodeRef<Expression>, FieldId> columnReferences = analysis.getColumnReferenceFields();
 
         return AstUtils.preOrder(node)
                 .filter(Expression.class::isInstance)
                 .map(Expression.class::cast)
                 .filter(expression -> columnReferences.containsKey(NodeRef.of(expression)))
-                .filter(expression -> isReferenceToScope(expression, scope, columnReferences));
+                .filter(expression -> hasReferenceToScope(expression, scope, columnReferences));
     }
 
-    private static boolean isReferenceToScope(Expression node, Scope scope, Map<NodeRef<Expression>, FieldId> columnReferences)
+    private static boolean hasReferenceToScope(Expression node, Scope scope, ListMultimap<NodeRef<Expression>, FieldId> columnReferences)
     {
-        FieldId fieldId = columnReferences.get(NodeRef.of(node));
-        requireNonNull(fieldId, () -> "No FieldId for " + node);
-        return isFieldFromScope(fieldId, scope);
+        return columnReferences.get(NodeRef.of(node)).stream().anyMatch(fieldId -> isFieldFromScope(fieldId, scope));
```

Comment:
Shouldn't it be allMatch?

Reply:
This function is finding references to the given scope, so as long as there is some reference (any), we should include it.